src/app/route/SearchPage/SearchPage.component.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * ScandiPWA - Progressive Web App for Magento
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/base-theme
+ * @link https://github.com/scandipwa/base-theme
+ */
+
+import PropTypes from 'prop-types';
+import CategoryPage from 'Route/CategoryPage/CategoryPage.component';
+import {
+    getUrlParam, getQueryParam, setQueryParams, clearQueriesFromUrl
+} from 'Util/Url';
+
+
+class SearchPage extends CategoryPage {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            sortKey: 'name',
+            sortDirection: 'ASC',
+            defaultPriceRange: { min: 0, max: 300 },
+            minPriceRange: 0,
+            maxPriceRange: 300,
+            previousPage: 0,
+            pageSize: 12
+        };
+    }
+
+    componentWillMount() {
+        this.updateBreadcrumbs();
+    }
+
+    componentDidMount() {
+        const { isOnlyPlaceholder, updateLoadStatus } = this.props;
+
+        if (!isOnlyPlaceholder) {
+            this.updateBreadcrumbs();
+            this.requestCategory();
+        } else {
+            updateLoadStatus(true);
+        }
+    }
+
+    componentDidUpdate(prevProps) {
+        const { location } = this.props;
+
+        if (this.urlHasChanged(location, prevProps)) {
+            this.updateBreadcrumbs();
+            this.requestCategory();
+        }
+    }
+
+    /**
+     * Prepare and dispatch category request
+     * @return {void}
+     */
+    requestCategory() {
+        const {
+            requestCategory,
+            location,
+            isSearchPage,
+            items,
+            category,
+            categoryIds
+        } = this.props;
+        const {
+            sortKey,
+            sortDirection,
+            previousPage,
+            pageSize,
+        } = this.state;
+        const categoryUrlPath = !categoryIds ? this.getCategoryUrlPath() : null;
+        const currentPage = getQueryParam('page', location) || 1;
+        const priceRange = this.getPriceRangeFromUrl();
+        const customFilters = this.getCustomFiltersFromUrl();
+        const querySortKey = getQueryParam('sortKey', location);
+        const querySortDirection = getQueryParam('sortDirection', location);
+        const options = {
+            search: getUrlParam({ path: 'search/' }, location),
+            isSearchPage: isSearchPage || false,
```

Comment:
What does this does? Can we please not use this `isSearchPage`? It looks like a hack to me.